### PR TITLE
Add support for getting a string representation of the LayerTree

### PR DESCRIPTION
**Pull request by gw280**
## Description

This adds a method (based on getRenderTree) which allows developers to programmatically get a string representation of the layer tree using flutter_driver.

## Related Issues

https://github.com/flutter/flutter/issues/46992

## Tests

I added the following tests:

`flutter_driver_test.dart`: getLayerTree

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

### CODE REVIEW

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

File: packages/flutter_driver/lib/src/extension/extension.dart [hunk 1]
```diff
@@ -229,6 +232,10 @@ class FlutterDriverExtension {
 
   Future<Health> _getHealth(Command command) async => const Health(HealthStatus.ok);
 
+  Future<LayerTree> _getLayerTree(Command command) async {
+    return LayerTree(RendererBinding.instance?.renderView?.debugLayer?.toStringDeep());
```
Comment (dnfield) [inline packages/flutter_driver/lib/src/extension/extension.dart]: This will not work in profile/release mode - is that ok?

Comment (dnfield) [reply packages/flutter_driver/lib/src/extension/extension.dart]: (or rather, it will return `null` in those cases)

**Review by dnfield — APPROVED** [review]
LGTM with comment

File: packages/flutter_driver/lib/src/extension/extension.dart (hunk 1 above)
Comment (gw280) [reply packages/flutter_driver/lib/src/extension/extension.dart]: I think it's fine, this should match the existing behaviour (https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/binding.dart#L448)

**Pull request merged**